service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -233,6 +242,37 @@ public static void deleteSiteImage(String imageUrl) {
     }
   }
 
+  /**
+   * Load the existing site image with the given URL from the user uploads S3 bucket.
+   *
+   * @param imageUrl the URL of the image file in S3 to delete.
+   * @throws SdkClientException if the load from S3 failed.
+   */
+  public static AttachmentResource loadS3Image(String imageUrl) {
+    String imagePath = imageUrl.split(externs.getBucketPublicUrl() + '/')[1];
+
+    if (!pathExists(imagePath)) {
+      throw new InvalidURLException();
+    }
+
+    // Create the request to get the HTML
+    GetObjectRequest awsRequest = new GetObjectRequest(externs.getBucketPublic(), imagePath);
+
+    S3Object image = externs.getS3Client().getObject(awsRequest);
+    S3ObjectInputStream is = image.getObjectContent();
+
+    String mimeType = "img/" + image.getObjectMetadata().getContentType();
+    DataSource datasource;
+    try {
+      datasource = new ByteArrayDataSource(is, mimeType);
+    } catch (IOException e) {
+      throw new MalformedParameterException("Image encoding is incompatible");
+    }
+
+    String name = image.getKey();
+    return new AttachmentResource(name, datasource);
+  }
```

Comment:
nit: IMO I think this part of the method should live somewhere else (probably in the `loadSiteImage()` you added here), and have this method just return the S3object we fetched. The `S3Requester` class should be concerned only about doing S3 stuff, and converting an image into an emailer-specific attachment class should fall under the `Emailer`'s concerns. 

This separation of concerns keeps our classes and methods logically organized and reusable for the future - if we ever wanted to fetch an object from S3 and do something else besides turn it into an `AttachmentResource`, we'll already have a nice method to use,

Reply:
I agree, something felt a little wrong about the organization so thanks for spelling it out.